### PR TITLE
Restructure awwvision

### PR DESCRIPTION
**Pull request by mattysweeps**
Working towards #10

* Update root package to match other applications
* Moves classes into different sub-packages
* Rename service classes to be specific to their business logic
* Add `Controller` suffix to the controllers

### CODE REVIEW

File: awwvision/src/main/java/com/google/cloud/servicebroker/samples/awwvision/service/CuteImageService.java [hunk 1]
```diff
@@ -34,19 +34,19 @@
...
+public class CuteImageService {
 
   @Autowired
   private Storage storageService;
 
   private final String bucketName;
 
-  StorageApi() {
+  CuteImageService() {
     String env = System.getenv("VCAP_SERVICES");
```
Comment (josephlewis42) [inline awwvision/src/main/java/com/google/cloud/servicebroker/samples/awwvision/service/CuteImageService.java:50]: I'm not sure if this is needed if we have the right spring-boot-starter, I think that one is set up to work with storage buckets, right?

File: awwvision/src/main/java/com/google/cloud/servicebroker/samples/awwvision/controller/RedditScraperController.java [hunk 2]
```diff
@@ -88,11 +89,11 @@ void storeAndLabel(RedditResponse response) {
...
           // Only label and upload the image if it does not already exist in storage.
-          StorageObject existing = storageApi.get(listing.data.url);
+          StorageObject existing = cuteImageService.get(listing.data.url);
           if (existing == null) {
-            String label = visionApi.labelImage(raw);
+            String label = imageLabelingService.labelImage(raw);
             if (label != null) {
-              storageApi.uploadJpeg(listing.data.url, url, ImmutableMap.of("label", label));
+              cuteImageService.uploadJpeg(listing.data.url, url, ImmutableMap.of("label", label));
             }
```
Comment (josephlewis42) [inline awwvision/src/main/java/com/google/cloud/servicebroker/samples/awwvision/controller/RedditScraperController.java]: what do you think about refactoring this to this? I think it'll make it cleaner, and reduce the useless work we do on our outbound network calls:

```
void storeAndLabel(RedditResponse response) {
  for (Listing listing : response.data.children) {
    storeAndLabel(listing); // functional support?
  }
}

private void storeAndLabel(final Listing listing) {
  Objects.requireNonNull(listing, "listing must not be null");
  if (listing.data.preview == null) { // is this a bug that should be listing.data.url?
    return;
  }
  
  final String dataUrl = listing.data.url;
  try {
    // Only label and upload the image if it does not already exist in storage.
    final StorageObject existing = cuteImageService.get(dataUrl);
    if (existing != null) {
      return;
    }
    
    final URL url = new URL(dataUrl);
    final byte[] raw = download(url);
    final String label = imageLabelingService.labelImage(raw);
    if (label != null) {
      cuteImageService.uploadJpeg(dataUrl, url, ImmutableMap.of("label", label));
    }
  } catch (IOException ex) {
    logger.error("Issue with labeling or uploading image " + dataUrl, ex);
  }
}
```

File: awwvision/src/main/java/com/google/cloud/servicebroker/samples/awwvision/service/ImageLabelingService.java [hunk 3]
```diff
@@ -29,10 +29,10 @@
 import java.io.IOException;
 
 /**
- * Helper methods for interacting with the Cloud Vision API.
+ * Component which labels images using Google Vision.
```
Comment (josephlewis42) [inline awwvision/src/main/java/com/google/cloud/servicebroker/samples/awwvision/service/ImageLabelingService.java]: I think our branding guidelines say the product name needs to be "Cloud Vision API"

**Review by josephlewis42 — CHANGES_REQUESTED** [review]
overall, I think the refactoring is a great thing, makes this app very business-y! Just a few comments.

File: awwvision/src/main/java/com/google/cloud/servicebroker/samples/awwvision/service/CuteImageService.java (hunk 1 above)
Comment (mattysweeps) [reply awwvision/src/main/java/com/google/cloud/servicebroker/samples/awwvision/service/CuteImageService.java:50]: Yup, I was going to do that in another commit. I'll add it to this PR

**Review by josephlewis42 — APPROVED** [review]

**Comment by josephlewis42** [comment]
LGTM, if you want to do the spring boot starter change in another PR that's fine with me.

**Pull request merged**